content/docs/pipelines/sdk/static-type-checking.md

```diff
@@ -0,0 +1,144 @@
++++
+title = "DSL Static Type Checking"
+description = "Statically check the component I/O types"
+weight = 6
++++
+
+## Overview
+A pipeline is a workflow consisting of [components](/docs/pipelines/sdk/build-component#overview-of-pipelines-and-components) and each
+component contains inputs and outputs. The DSL compiler supports static type checking to ensure the type consistency among the component
+I/Os within the same pipeline. Static type checking helps users to identify component I/O inconsistencies without running the pipeline. 
+It also shortens the development cycles by catching the errors early. 
+This feature is especially useful in two cases: 
+* When the pipeline is huge and manually checking the types is infeasible; 
+* When some components are shared ones and the type information is not immediately available to the pipeline authors.
+
+## Type System  
+In Kubeflow pipeline, a type is defined as a type name with an [OpenAPI Schema](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md)
+property, which defines the input parameter schema. *Warning*( the pipeline system 
+currently does not check the input value against the shcema when users submit a pipeline run. However, this feature will come in the near 
+future). There is a set of [core types](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_types.py) defined in the 
```

Comment:
Remove the closing parenthesis ")" after "future". as there's now no opening one.

Reply:
done.